File: rust/src/models/integration_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct IntegrationOut {
    pub name: String,

    /// The Integration's ID.
    pub id: String,

    #[serde(rename = "createdAt")]
    pub created_at: chrono::DateTime<chrono::Utc>,

    #[serde(rename = "updatedAt")]
    pub updated_at: chrono::DateTime<chrono::Utc>,

    /// The set of feature flags the integration has access to.
    #[serde(rename = "featureFlags")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub feature_flags: Option<std::collections::BTreeSet<String>>,
}

impl IntegrationOut {
    pub fn new(
        name: String,
        id: String,
        created_at: chrono::DateTime<chrono::Utc>,
        updated_at: chrono::DateTime<chrono::Utc>,
    ) -> Self {
        Self {
            name,
            id,
            created_at,
            updated_at,
            feature_flags: None,
        }
    }
}

File: javascript/src/models/integrationOut.ts
// this file is @generated

export interface IntegrationOut {
  name: string;
  /** The Integration's ID. */
  id: string;
  createdAt: Date;
  updatedAt: Date;
  /** The set of feature flags the integration has access to. */
  featureFlags?: string[];
}

export const IntegrationOutSerializer = {
  _fromJsonObject(object: any): IntegrationOut {
    return {
      name: object["name"],
      id: object["id"],
      createdAt: new Date(object["createdAt"]),
      updatedAt: new Date(object["updatedAt"]),
      featureFlags: object["featureFlags"],
    };
  },

  _toJsonObject(self: IntegrationOut): any {
    return {
      name: self.name,
      id: self.id,
      createdAt: self.createdAt,
      updatedAt: self.updatedAt,
      featureFlags: self.featureFlags,
    };
  },
};
